From 756d9c67b5aa04ae83912188a2b86844667703d5 Mon Sep 17 00:00:00 2001
Subject: [PATCH 03/13] Be a little more precise with our types in read_file()

Based on the bug and patch reported in this thread:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=764386

Signed-off-by: Peter Jones <pjones@redhat.com>
Signed-off-by: Curt Brune <curt@cumulusnetworks.com>

diff --git a/src/util.h b/src/util.h
index 5fa01a0..94708e1 100644
--- a/src/util.h
+++ b/src/util.h
@@ -31,7 +31,8 @@ read_file(int fd, uint8_t **buf, size_t *bufsize)
 {
 	uint8_t *p;
 	size_t size = 4096;
-	int s = 0, filesize = 0;
+	size_t filesize = 0;
+	ssize_t s = 0;
 
 	uint8_t *newbuf;
 	if (!(newbuf = calloc(size, sizeof (uint8_t))))
